### PR TITLE
Recover missing state after collaborator restart

### PR DESCRIPTION
**Pull request by cloudnoize**
After restart the collaborator gets a list of tasks that it needs to finish as part of the current round, that list might be partial as it already executed some tasks prior the restart, previous tasks might have generated state that is needed for the future tasks of this round.
This PR recover or initialized the state of previous tasks enabling the collaborator to resume the execution of the current round.

https://jira.devtools.intel.com/browse/FEDAI-1229

### CODE REVIEW

File: openfl/federated/task/runner_keras.py [hunk 1]
```diff
@@ -182,7 +182,16 @@ def train_(self, batch_generator, metrics: list = None, **kwargs):
...
-        results = self.model.get_metrics_result()
+        try:
+            results = self.model.get_metrics_result()
+        except ValueError:
+            if "batch_size" in kwargs:
+                batch_size = kwargs["batch_size"]
+            else:
+                batch_size = 1
+            # evaluation needed before metrics can be resolved
+            self.model.evaluate(self.data_loader.get_valid_loader(batch_size), verbose=1)
```
Comment (psfoley) [inline openfl/federated/task/runner_keras.py:193]: For other reviewers reference, this is needed for the scenario when the collaborator dies between the `aggregated_model_validation` and `train` task. If the collaborator restarts it will immediately be assigned `train`, and will not be able to resolve the necessary metrics. The names of needed metrics can be resolved by running the `evaluate` function on the model, but these metric values will not be sent to the aggregator.

**Review by psfoley — APPROVED** [review]

**Review by rahulga1 — APPROVED** [review]

**Review by tanwarsh — APPROVED** [review]
